borg/crypto.pyx

```diff
@@ -74,6 +74,14 @@ bytes_to_long = lambda x, offset=0: _long.unpack_from(x, offset)[0]
 long_to_bytes = lambda x: _long.pack(x)
 
 
+class CryptoError(Exception):
+    """Malfunction in the crypto module."""
+
+
+class IntegrityError(CryptoError):
+    """Integrity checks failed. Corrupted or tampered data."""
```

Comment:
i wanted to avoid potentially cyclic imports and not import stuff from higher-level python modules into lower-level cython stuff, but yes, in the end 1 is enough.